syntax = "proto3";

package grpc;

option go_package = "./grpc";

message ArticleEmbeddingRequest {
  int32 id = 1;
  string title = 2;
  string content = 3;
}

message ArticleEmbeddingResponse {
}

message SearchRequest {
  string query = 1;
  int32 limit = 2;
}

message SearchResponse {
  repeated int32 article_ids = 1;
}

service SearchService {
  rpc IndexArticle(ArticleEmbeddingRequest) returns (ArticleEmbeddingResponse);
  rpc SemanticSearch(SearchRequest) returns (SearchResponse);
}